src/documentation/search/search-hooks.tsx

```diff
@@ -22,15 +39,53 @@ export const useSearch = (): Search => {
 
 const SearchProvider = ({ children }: { children: ReactNode }) => {
   const { exploreToolkit, referenceToolkit } = useToolkitState();
+  const [query, setQuery] = useState<string>("");
+  const [results, setResults] = useState<SearchResults | undefined>();
+  const isUnmounted = useIsUnmounted();
+  const logging = useLogging();
   useEffect(() => {
     // Wait for both, no reason to index with just one then redo with both.
     if (exploreToolkit.status === "ok" && referenceToolkit) {
       search.index(exploreToolkit.toolkit, referenceToolkit);
     }
   }, [exploreToolkit, referenceToolkit]);
 
+  const debouncedSearch = useMemo(
+    () =>
+      debounce(async (newQuery: string) => {
+        const trimmedQuery = newQuery.trim();
+        if (trimmedQuery) {
+          const results = await search.search(trimmedQuery);
+          if (!isUnmounted()) {
+            setResults((prevResults) => {
+              if (!prevResults) {
+                logging.event({ type: "search" });
+              }
+              return results;
+            });
+          }
+        } else {
+          setResults(undefined);
+        }
+      }, 300),
+    [setResults, isUnmounted, logging]
+  );
+
+  const [{ languageId }] = useSettings();
+  useEffect(() => {
+    setQuery("");
+  }, [languageId]);
+
+  useEffect(() => {
+    debouncedSearch(query);
```

Comment:
Something like this prevents the delay in closing the results on clearing the search input.
```suggestion
    if (!query.trim()) {
      setResults(undefined);
    } else {
      debouncedSearch(query);
    }
```

Reply:
[`[6d15abb](https://github.com/microbit-foundation/python-editor-next/pull/574/commits/6d15abb74b7269c1afd42518e6c0246210169ba1)`](https://github.com/microbit-foundation/python-editor-next/pull/574/commits/6d15abb74b7269c1afd42518e6c0246210169ba1)